src/client/app/containers/LineChartContainer.ts

```diff
@@ -4,28 +4,39 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-import { Line } from 'react-chartjs-2';
-import moment from 'moment';
+import * as _ from 'lodash';
+import { Line, ChartComponentProps } from 'react-chartjs-2';
+import { ChartData, ChartDataSets, ChartPoint, ChartTooltipItem } from 'chart.js';
+import * as moment from 'moment';
 import { connect } from 'react-redux';
 import getGraphColor from '../utils/getGraphColor';
+import { State } from '../types/redux';
 
 /**
  * @param {State} state
  */
-function mapStateToProps(state) {
+function mapStateToProps(state: State) {
 	const timeInterval = state.graph.timeInterval;
-	const data = { datasets: [] };
+	const datasets: ChartDataSets[] = [];
 
 	// Add all meters data to the chart
 	for (const meterID of state.graph.selectedMeters) {
 		const byMeterID = state.readings.line.byMeterID[meterID];
 		if (byMeterID !== undefined) {
-			const readingsData = byMeterID[timeInterval];
+			const readingsData = byMeterID[timeInterval.toString()];
 			if (readingsData !== undefined && !readingsData.isFetching) {
 				const label = state.meters.byMeterID[meterID].name;
-				data.datasets.push({
+				if (readingsData.readings === undefined) {
+					throw new Error('Unacceptable condition: readingsData.readings is undefined.');
```

Comment:
Perhaps instead notify the user that an error occured. 99% of users won't see error logs in the console

Reply:
Yeah, when we merge your PR that makes the better notification system I plan to transition to that.